File: golang/Level 1/약수의_합.go
func solution(n int) int {
    if n == 0 {
        return 0
    }
    
    var temp, p int
    
    answer := 1
    for d := 2; n != 1; d++ {
        temp = 1
        p = 1
        for n % d == 0 {
            temp += p * d
            p *= d
            n = int(n / d)
        }
        answer *= temp
    }
    return answer
}

File: python/Level 1/약수의_합.py
def solution(n):
    if n == 0:
        return 0

    answer, d = 1, 2
    while n != 1:
        temp, p = 1, 1
        while n % d == 0:
            temp += p * d
            p *= d
            n //= d
            
        answer *= temp
        d += 1

    return answer
